jni/include/native_engines_stream_support.h

```diff
@@ -119,6 +121,110 @@ class NativeEngineIndexInputMediator {
 
 
 
+/**
+ * This class delegates the provided index output to do IO processing.
+ * In most cases, it is expected that IndexOutputWithBuffer was passed down to this,
+ * which eventually have Lucene's IndexOutput to write bytes.
+ */
+class NativeEngineIndexOutputMediator {
```

Comment:
Please add clearer comments to describe the purpose and functionality of each member and method. The initialization of jvalue args using the union syntax, for example, should have a clearer explanation.

Reply:
hmm.. yeah, it's always better to have a clearer explanation attach to it...
but not sure what is missing.
could you pinpoint what is missing form the comment?? I think it provide a general view of the class

Reply:
Yeah one instance was I was not aware of  jvalue args using the union syntax, if we could see something helps understand on special syntax , would be better